cpp/include/cudf/column/column.hpp

```diff
@@ -347,6 +347,18 @@ class column {
    */
   operator mutable_column_view() { return this->mutable_view(); };
 
+  /**
+   * @brief Sets the stream to be used for deallocation of internal buffers
+   */
+  void set_stream(rmm::cuda_stream_view stream) noexcept
+  {
+    _data.set_stream(stream);
+    _null_mask.set_stream(stream);
+    for (auto& child : _children) {
+      child->set_stream(stream);
+    }
+  }
```

Comment:
Would prefer the function definition be put into `cpp/src/column/column.cu` instead of in the header.

Reply:
This has changed to a templated function, but I pulled out the child functions.